File: src/jp/lionas/alexa/fundra/AnswerYesIntentHandler.java
package jp.lionas.alexa.fundra;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.*;
import com.google.gson.Gson;
import jp.lionas.alexa.fundra.mufg.FundUtil;
import jp.lionas.alexa.fundra.mufg.MufgFund;
import jp.lionas.alexa.fundra.mufg.model.CodeItem;

import java.util.*;

import static com.amazon.ask.request.Predicates.intentName;
import static com.amazon.ask.request.Predicates.sessionAttribute;

public class AskYesIntentHandler implements RequestHandler {

    @Override
    public boolean canHandle(HandlerInput handlerInput) {
        return handlerInput.matches(intentName(Const.ASK_YES_INTENT_NAME)
                .and(sessionAttribute(Const.STATE_KEY, Const.STATE_REPEAT)));
    }

    @Override
    public Optional<Response> handle(HandlerInput handlerInput) {

        Map<String, Object> sessionAttributes = handlerInput.getAttributesManager().getSessionAttributes();
        String fundName = (String)sessionAttributes.get(Const.FUND_NAME);
        List<CodeItem> found = MufgFund.findFund(fundName);

        // 複数マッチした場合
        return FundUtil.getRepeatedFunds(handlerInput, sessionAttributes, found);

    }


}

File: src/jp/lionas/alexa/fundra/AnswerYesIntentHandler.kt
package jp.lionas.alexa.fundra

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.dispatcher.request.handler.RequestHandler
import com.amazon.ask.model.Response
import com.amazon.ask.request.Predicates.intentName
import com.amazon.ask.request.Predicates.sessionAttribute
import jp.lionas.alexa.fundra.def.State
import jp.lionas.alexa.fundra.mufg.FundUtil
import jp.lionas.alexa.fundra.mufg.MufgFund
import java.util.*

class AnswerYesIntentHandler : RequestHandler {

    override fun canHandle(handlerInput: HandlerInput): Boolean {
        return handlerInput.matches(intentName("AMAZON.YesIntent")
                .and(sessionAttribute(State.STATE_KEY, State.STATE_REPEAT)))
    }

    override fun handle(handlerInput: HandlerInput): Optional<Response> {

        val sessionAttributes = handlerInput.attributesManager.sessionAttributes
        val fundName = sessionAttributes[State.FUND_NAME] as String
        val found = MufgFund.findFund(fundName)

        // 複数マッチした場合
        return FundUtil.getRepeatedFunds(handlerInput, sessionAttributes, found)

    }


}
